mailreply00.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2023 Matthew Holt
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package acmez
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/mholt/acmez/acme"
+)
+
+// MailChallangeReplyGen builds email body with headers to reply MailReply00
+// challange Response email. This fucntion just build body, and sendung
+// message have to done by caller of this function
+// mailSubject and messageId is such from challange mail,
+// and if there is no reply-to header in challange email, replyto parametere should be ""
+func MailChallangeReplyGen(c acme.Challenge, mailSubject string, messgageId string, replyto string) string {
+	if len(replyto) == 0 {
+		replyto = c.From
+	}
+	mailSubject = strings.TrimPrefix(mailSubject, "ACME: ")
+	keyauth := c.MailReply00KeyAuthorization(mailSubject)
```

Comment:
IMO it's more logical to put the full token (part 1 and 2) into the key authorization; not just the part that's in the email. By putting the entire subject in here, the `Re: ACME: <part1>` will be different for the response (i.e. `Re: ACME: <part1><part2>`). The spec isn't explicit about this, but from an email view point, changing the subject in the `Re` is not the usual behavior. Also see my remark on MailReply00KeyAuthorization.

Reply:
Yeah, I might've misread the code the first time, or some logic changed in the meantime. As long as it says `"Re: ACME: <base64-url(token-part1)>"`, it seems OK.